src/main/java/network/brightspots/rcv/Tabulator.java

```diff
@@ -206,6 +206,13 @@ Set<String> tabulate() throws TabulationCancelledException {
         // 2. If there's a minimum vote threshold, drop all candidates below that threshold.
         if (eliminated.isEmpty()) {
           eliminated = dropCandidatesBelowThreshold(currentRoundTallyToCandidates);
+          // One edge case: if everyone is below the threshold, we can't proceed. This would only
+          // happen in the first or (if we drop undeclared write-ins first) second round.
+          if (eliminated.size() == config.getNumDeclaredCandidates()) {
+            Logger.severe("Tabulation can't proceed because all declared candidates are below "
+                + "the minimum vote threshold.");
+            throw new TabulationCancelledException(false);
```

Comment:
It's not really cancelled, right?  It's just ending sooner than we thought.

Reply:
Would you be more on board with "aborted" instead of "cancelled"? I think that captures both situations where this exception is used (giving up because no one meets the threshold or exiting because the user clicked a Cancel button). Just `TabulationException` is also OK, but I'd prefer to be more specific.

Reply:
Sure, sounds good.